### PR TITLE
build: fix GLIBC < 2.18 compatible issues and android arm binary loading

### PR DESCRIPTION
**Pull request by Brooooooklyn**
- Close https://github.com/swc-project/swc/issues/1723
- Close https://github.com/swc-project/swc/issues/3198

### Note
- `rust-toolchain` is upgraded to `nightly-2022-01-19` to avoid `aarch64-linux` cross compile issue.
- `swc_cli` is built as `linux-aarch64-musl` in `linux-aarch64-gnu` build step. Because `musl` executable binary is also works in `gnu` platform.

### CODE REVIEW

**Comment by Austaras** [comment]
Seems to be still blocked by https://github.com/swc-project/swc/pull/3196#discussion_r778537610

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2021-11-15
\ No newline at end of file
+nightly-2022-01-19
```
[the remark this answers is not in the document]
Comment (Brooooooklyn) [reply rust-toolchain:1]: How could I test if it works?

[the remark this answers is not in the document]
Comment (Brooooooklyn) [reply rust-toolchain:1]: Running `cargo build -p swc --features plugin --release` successfully on Linux

File: crates/swc_visit_macros/src/lib.rs [hunk 2]
```diff
@@ -1568,28 +1568,14 @@ fn skip(ty: &Type) -> bool {
         Type::Path(p) => {
             let i = &p.path.segments.last().as_ref().unwrap().ident;
 
-            if i == "bool"
-                || i == "u128"
+            p.path.segments.last().as_ref().unwrap().ident == "bool"
```
Comment (kdy1) [inline crates/swc_visit_macros/src/lib.rs:1571]: This seems bit strange

**Review by kdy1 — APPROVED** [review]
Thanks! I'll invert it when required.

File: rust-toolchain (hunk 1 above)
[the remark this answers is not in the document]
Comment (magic-akari) [reply rust-toolchain:1]: Could someone help me?
I ran into this error:
```
error: the 'cargo' binary, normally provided by the 'cargo' component, is not applicable to the 'nightly-2022-01-19-aarch64-apple-darwin' toolchain
```

[the remark this answers is not in the document]
Comment (kdy1) [reply rust-toolchain:1]: I'm not sure why it happens.

https://rust-lang.github.io/rustup-components-history/aarch64-apple-darwin.html says there's `cargo` in `2022-01-19-aarch64-apple-darwin`.

[the remark this answers is not in the document]
Comment (magic-akari) [reply rust-toolchain:1]: > I'm not sure why it happens.
> 
> https://rust-lang.github.io/rustup-components-history/aarch64-apple-darwin.html says there's `cargo` in `2022-01-19-aarch64-apple-darwin`.

Thanks, fixed by cleaning the rust toolchain.

**Comment by kdy1** [comment]
Semes like `STACK_BUFFER_OVERRUN` is not fixed yet? https://github.com/swc-project/swc/runs/4877028875?check_suite_focus=true

**Comment by Brooooooklyn** [comment]
I'll try to fix it. It's randomly appeared before. We should adjust some LTO options for Windows arm64 to let the linker consume less memory.

**Comment by kdy1** [comment]
Ah, thank you!
I tried downgrading `rustc`, but it didn't work. Can you please bump it again with the PR for fix?

**Comment by Brooooooklyn** [comment]
okay

**Comment by Brooooooklyn** [comment]
> I tried downgrading rustc, but it didn't work. 

Seems like it is just because our project is too large for a `7GB RAM` machine to build.

**Comment by kdy1** [comment]
Oh... I see. I'll investigate

**Pull request merged**